src/db/tables/subscribers.js

```diff
@@ -271,6 +272,59 @@ async function deleteResolutionsWithEmail (id, email) {
 }
 /* c8 ignore stop */
 
+/**
+ * @param {Partial<{ plusOnly: boolean; limit: number; }>} options
+ * @returns {Promise<import("knex/types/tables").SubscriberRow[]>}
+ */
+// Not covered by tests; mostly side-effects. See test-coverage.md#mock-heavy
+/* c8 ignore start */
+async function getSubscribersWaitingForMonthlyEmail (options = {}) {
+  // I'm explicitly referencing the type here, so that these lines of code will
+  // show up as errors when we remove it from the flag list:
+  /** @type {import("./featureFlags.js").FeatureFlagName} */
+  const featureFlagName = "MonthlyActivityEmail";
+  const flag = await knex("feature_flags")
+    .first()
+    .where("name", featureFlagName);
+
+  if (!flag?.is_enabled) {
+    return [];
+  }
+
+  let query = knex('subscribers')
+    .select()
+    .where((builder) => builder.whereNull("monthly_email_optout").orWhere("monthly_email_optout", false))
+    .andWhere(builder => builder.whereNull("monthly_email_at").orWhereRaw('"monthly_email_at" < NOW() - INTERVAL \'30 days\''));
+
+  if (Array.isArray(flag.allow_list) && flag.allow_list.length > 0) {
+    // The `.andWhereIn` alias doesn't exist:
+    // https://github.com/knex/knex/issues/1881#issuecomment-275433906
+    query = query.whereIn("primary_email", flag.allow_list)
+  }
+
+  if (options.plusOnly) {
+    // Note: This will only match people who had a Plus subscription the last
+    //       time they signed in. We'd have to check with SubPlat directly to
+    //       find out if they have a subscription right now.
+    query = query.andWhereRaw(
+      `EXISTS (
+        SELECT 1
+        FROM jsonb_array_elements(fxa_profile_json->'subscriptions') AS subscription
+        WHERE subscription::text = '"${MONITOR_PREMIUM_CAPABILITY}"'
```

Comment:
I know it's a `const` in this case but I'd still use a parameterized query here, in case anything changes in the future. You should be able to use it with raw queries as well in knex.

Reply:
Oh yes absolutely we should, better safe than sorry: 39704eb9f